app/models/compute_resources/foreman/model/vmware.rb

```diff
@@ -816,8 +822,14 @@ def vm_instance_defaults
     end
 
     def firmware_mapping(firmware_type)
-      return 'efi' if firmware_type == :uefi
-      'bios'
+      case firmware_type
+      when :uefi
+        'efi'
+      when :uefi_sb
+        'uefi_sb'
+      else
+        'bios'
+      end
     end
```

Comment:
Same here, consider moving this method to the `ComputeResource` model to be used by both VMware and Libvirt.